compiler/rustc_errors/src/lib.rs

```diff
@@ -661,6 +661,19 @@ impl Handler {
         inner.stashed_diagnostics = Default::default();
     }
 
+    /// Avoid failing compilation in the presence of delayed bugs.
+    ///
+    /// NOTE: *do not* call this function from rustc. It is only meant to be called from rustdoc.
+    /// Rustdoc documents multiple targets at once, meaning there may be multiple versions
+    /// of the same function in scope at the same time, which isn't legal Rust otherwise. See
+    /// <https://doc.rust-lang.org/beta/rustdoc/advanced-features.html#interactions-between-platform-specific-docs>
+    /// for details
+    pub fn reset_delayed_bugs(&self) {
```

Comment:
Would it be possible to still keep a record of the fact that there was a delayed bug and if so ICE when using `sess.compile_error()` but ignore this flag when using a new `sess.compile_error_allow_reset_delay_bugs()` to be used in rustdoc? Or maybe even have that later method be the method that is actually responsible for ignoring delayed bugs?